### PR TITLE
[DE-2699] fix values stg_pashim__policies

### PR DESCRIPTION
**Pull request by jennaallen**
Details here https://vouchinc.atlassian.net/browse/DE-2699

I also setup CI, with Nabeel's help, to run the integration tests as part of each PR

### CODE REVIEW

File: macros/as_primitive.sql [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
-        select 
-            typeof(" ~ c ~ ") as type, 
-            ifnull(len(split(" ~ c ~ ", '.')[1]), 2) as precision 
+        select
+            ifnull(try_parse_json(" ~ c ~ "), " ~ c ~ ") as parsed,
+            typeof(parsed) as type, 
+            ifnull(len(split(parsed, '.')[1]), 2) as scale 
         from " ~ t ~ " 
-        where type is not null 
+        where ifnull(type, 'NULL_VALUE') != 'NULL_VALUE'
```
Comment (avyfain) [inline macros/as_primitive.sql:11]: is the point here that we can get both the SQL null and the JSON null?

**Review by avyfain — APPROVED** [review]
This makes sense to me, just one question

File: macros/as_primitive.sql (hunk 1 above)
Comment (jennaallen) [reply macros/as_primitive.sql:11]: Good question. I believe the answer is yes. Basically, you can get a scenario where the key `c` exists in the JSON, but the value is `null` and in that case you get a type of `NULL_VALUE`. In the other scenario the key `c` doesn't exists in the JSON and in that case the type = `null`

Comment (avyfain) [reply macros/as_primitive.sql:11]: let's make a comment about that, since it's not obvious, then merge?

Comment (jennaallen) [reply macros/as_primitive.sql:11]: Will do

**Pull request merged**